### DISCUSSION TITLE
Trouble with recording on the hour consistently

### DISCUSSION THREAD

**Discussion by caden-fluegge** [gopro/labs#878] [Q&A] [answered]
Hello,

I posted a couple of weeks ago about getting a Hero 12 to record one minute every hour between 7 am and 7 pm.  I've started having issues with the camera displaying "Labs Processing" across the top of the rear screen, leading to the camera remaining on for periods between recording instead of going to sleep.  Here are the two most recent strings I've tried:

`!3540N<19:00!S+!56E~!06:00NQ!R`  (06:00 because of hour sleep before first video)

`<19:00!S+!56E~!07:00NQ!3540N!R`

Is there a better way to do this?  Could it be using an if/else is causing the issue (I hadn't noticed this problem when using 2 if statements, but the camera would record through the night, ignoring my time parameters)? Potentially setting 12 separate times for the camera to record?  I only ask this because we are most likely going to use our setup to have the camera record 3 times a day as well, for 3 minutes at a time (think 08:00, 12:00, 16:00), for long deployments offshore where consistency is key.

Thank you for any insight.

This is what I am trying next, with two if statements:
`!3540N>06:00<19:00!S+!56E+!R!06:00NQ!R`

-Caden

**Comment by dnewman-gpsw** [comment, marked as answer]
All the will likely work with a small modification.  I have found that !E end command is not always ready to do the next command in the script. Simple workaround: add !1N after the !E command. This adds an extra pause, before the shutdown inducing !3540N

Try this:
```<19:00!S+!56E~!07:00R!1N!3539R```

**Reply by caden-fluegge** [reply]
Giving this a go.  Thank you.  Is an exclamation point not required prior to the repeat at the end?  If `!1N` induces a 1 second pause, could this be extended with different values?  Do you know if there would be any benefit or detriment  to extending the pause duration to give the camera time to catch up? Just thinking for redundancy and increased reliability traded for minimal power loss.

**Reply by dnewman-gpsw** [reply]
Yes !2N thru !9N are all non-shutdown pauses.  Add !N is a 0.5s pause.

**Reply by dnewman-gpsw** [reply]
I believe !1N is plenty, as !N should work too.
